src/main/java/com/hackclub/hccore/listeners/SleepListener.java

```diff
@@ -158,7 +159,15 @@ private int getSleepingPlayers(World world) {
     }
 
     private int getMinSleepingPlayersNeeded(World world) {
+        //Get the number of AFK players
+        int afkSum = 0;
+        for (Player player : world.getPlayers()) {
+            PlayerData playerData = this.plugin.getDataManager().getData(player);
+            if (playerData.isAfk()) {
+                afkSum++;
+            }
+        }
         return (int) Math.ceil(world.getPlayers().size()
-                * this.plugin.getConfig().getDouble("settings.skip-sleep-threshold"));
+                * this.plugin.getConfig().getDouble("settings.skip-sleep-threshold")) - afkSum;
```

Comment:
you need to subtract `afkSum` before multiplying by `settings.skip-sleep-threshold`